x/sequencer/keeper/proposer.go

```diff
@@ -37,86 +36,47 @@ func (k Keeper) RollappPotentialProposers(ctx sdk.Context, rollappId string) []t
 	return append(seqs, k.SentinelSequencer(ctx))
 }
 
-// UpdateProposerIfNeeded will assign a proposer to the rollapp. It won't replace the incumbent proposer
-// if they are not sentinel. Otherwise it will prioritise a non sentinel successor. Finally, it
+// RecoverFromSentinelProposerIfNeeded will assign a proposer to the rollapp. It won't replace the incumbent proposer
+// if they are not sentinel. Otherwise it will prioritize a non sentinel successor. Finally, it
 // choose one based on an algorithm.
 // The result can be the sentinel sequencer.
-func (k Keeper) UpdateProposerIfNeeded(ctx sdk.Context, rollapp string) error {
+func (k Keeper) RecoverFromSentinelProposerIfNeeded(ctx sdk.Context, rollapp string) error {
 	proposer := k.GetProposer(ctx, rollapp)
-	before := proposer
 
 	// a valid proposer is already set so there's no need to do anything
 	if !proposer.Sentinel() {
 		return nil
 	}
```

Comment:
I find this function name and inner condition rather confusing as it tries to encapsulate various use cases and do multiple things in one function which make it non reusable. and from some call site (like the kick) I find it cumbersome. 

This function does:
1. check if new proposer is needed
2. set new proposer
3. set new successor 

for (1) - we can check it outside the function based on the callsite
for (2) - I think this all it should do
for (3) - we have the `setSuccessorForRollapp`. isn't it the same? shouldn't we just call it from callsite?


it's clearer to just make this function do one thing (i.e `recoverFromSentinel/UpdateProposer`) and simply check the condition and other work outside this function, where necessary. 

Currently this function is being called from 3 callsites:

1. create sequencer (need to check if current proposer is sentinel outside)
1. update sequencer (need to check if current proposer is sentinel and opted in before call)
1. kick (no need to check if current proposer is sentinel, as we're kicking)

Reply:
refactored